### PR TITLE
feat: add cloudflared spec

### LINKED ISSUE #1091: [cloudflared] Missing spec
### Sanity checks

- [X] My issue relates to a specific CLI completion spec (e.g. `git checkout` is missing options in `git` completion spec). If your issue is more general, please create your issue here: [withfig/fig](https://github.com/withfig/fig/issues/new?assignees=&labels=NEED_TO_LABEL&template=main_issue_template.yml)
- [X] I have searched [github.com/withfig/autocomplete/issues](https://github.com/withfig/autocomplete/issues?q=) and there are no duplicates of my issue
- [X] I would like to work on this.

### What CLI tool does this relate to?

https://github.com/cloudflare/cloudflared

### Which statement makes the most sense?

This completion spec does not exist

### Issue Details

Contains the command-line client for Cloudflare Tunnel, a tunneling daemon that proxies traffic from the Cloudflare network to your origins. This daemon sits between Cloudflare network and your origin (e.g. a webserver).

References: https://github.com/cloudflare/cloudflared#cloudflare-tunnel-client

### PR DESCRIPTION
**Pull request by diogofalken**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Feature.

**What is the current behavior? (You can also link to an open issue here)**
Closes #1091, missing spec of cloudflared.

**What is the new behavior (if this is a feature change)?**
Support for cloudflared CLI

### CODE REVIEW

**Comment by fedeci** [comment]
Was this generated through some tool or was it done manually?

**Comment by diogofalken** [comment]
> Was this generated through some tool or was it done manually?

@fedeci it was done manually.

**Comment by fedeci** [comment]
Thanks for all this work! It will be huge to review 😄

**Comment by diogofalken** [comment]
I am sorry, it was really a painful work 😅
Everything you see that I can improve just @ me, it's my first contribution 😁

File: src/cloudflared.ts [hunk 1]
```diff
@@ -0,0 +1,2410 @@
...
+      name: "--compression-quality",
+      description:
+        "(beta) Use cross-stream compression instead HTTP compression. 0-off, 1-low, 2-medium, >=3-high. (default: 0) [$TUNNEL_COMPRESSION_LEVEL]",
+      args: {
+        name: "value",
+        description: "Compression quality",
+        suggestions: [
+          { name: "0", description: "Off" },
+          { name: "1", description: "Low" },
+          { name: ">3", description: "High" },
```
Comment (fedeci) [inline src/cloudflared.ts:2341]: If we insert >3 the CLI will probably reject that value.
```suggestion
          { name: "3", description: "High" },
```

**Review by fedeci — CHANGES_REQUESTED** [review]
Thanks it looks really good! I left a few comments about thing to change. Great first contribution man🎉

File: src/cloudflared.ts [hunk 2]
```diff
@@ -0,0 +1,2410 @@
...
+        {
+          name: "--proxy-keepalive-connections --url",
+          description:
+            "HTTP proxy maximum keepalive connection pool size This flag only takes effect if you define your origin with --url and if you do not use ingress rules. The recommended way is to rely on ingress rules and define this property under `originRequest` as per https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/configuration/configuration-file/ingress (default: 100)",
+        },
+        {
+          name: "--proxy-keepalive-timeout --url",
+          description:
+            "HTTP proxy timeout for closing an idle connection This flag only takes effect if you define your origin with --url and if you do not use ingress rules. The recommended way is to rely on ingress rules and define this property under `originRequest` as per https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/configuration/configuration-file/ingress (default: 1m30s)",
+        },
```
[the remark this answers is not in the document]
Comment (diogofalken) [reply src/cloudflared.ts:1149]: That's true, do you think it's better to get it out and add a `dependsOn` on all of this?

File: src/cloudflared.ts [hunk 3]
```diff
@@ -0,0 +1,2410 @@
...
+          args: {
+            name: "value",
+            description: "Folder",
+            template: ["folders"],
+          },
+        },
+        {
+          name: "--trace-output",
+          description:
+            "Name of trace output file, generated when cloudflared stops. [$TUNNEL_TRACE_OUTPUT]",
```
[the remark this answers is not in the document]
Comment (diogofalken) [reply src/cloudflared.ts:1334]: I didn't know, just tried it out and it doesn't 🥲

File: src/cloudflared.ts [hunk 4]
```diff
@@ -0,0 +1,2410 @@
...
+    {
+      name: ["help", "h"],
+      description: "Shows a list of commands or help for one command",
+    },
+  ],
+  options: [
+    {
+      name: ["--help", "-h"],
+      description: "Show help (default: false)",
+    },
```
[the remark this answers is not in the document]
Comment (diogofalken) [reply src/cloudflared.ts]: Fixed in 27b7ed14ee2f094b815dd6b26406c16ef9ebca26.

File: src/cloudflared.ts [hunk 5]
```diff
@@ -0,0 +1,2410 @@
...
+      args: {
+        name: "url",
+      },
+    },
+    {
+      name: "curl",
+      description: "Uninstall the cloudflared launch agent",
+      options: [
+        {
+          name: ["--allow-request", "-ar"],
```
[the remark this answers is not in the document]
Comment (diogofalken) [reply src/cloudflared.ts]: It was a typo, thanks! Fixed in 27b7ed14ee2f094b815dd6b26406c16ef9ebca26

File: src/cloudflared.ts [hunk 6]
```diff
@@ -0,0 +1,2410 @@
...
+        "cloudflared tunnel [tunnel command options] create [subcommand options] NAME",
+      ],
+      options: [
+        {
+          name: "--config",
+          description:
+            'Specifies a config file in YAML format. (default: "/usr/local/etc/cloudflared/config.yml")',
+          args: {
+            name: "value",
+            description: "Config file",
```
[the remark this answers is not in the document]
Comment (diogofalken) [reply src/cloudflared.ts:394]: Added it on 27b7ed14ee2f094b815dd6b26406c16ef9ebca26

File: src/cloudflared.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (diogofalken) [reply src/cloudflared.ts]: Added `dependsOn` on all occurrences of this issue in 27b7ed14ee2f094b815dd6b26406c16ef9ebca26

**Comment by diogofalken** [comment]
@fedeci if you think something else needs to be changed @ me. Thank you :)

**Review by fedeci — APPROVED** [review]
Great! Thanks for this spec and I'm sorry for being late on merging this one!

**Pull request merged**